sigh/lib/assets/resign.sh

```diff
@@ -689,40 +695,45 @@ function resign {
             "com.apple.developer.associated-domains" \
             # If actually used by the App, this value will be set in its entitlements
             "com.apple.developer.default-data-protection" \
-            # PP seem to list the same groups as the App, but use App entitlements value to be sure
-            "com.apple.security.application-groups" \
             # Was already blacklisted in previous version, seems to be an artifact from an old Xcode release
             "com.apple.developer.maps" \
             # If actually used by the App, this value will be set in its entitlements
-            "com.apple.external-accessory.wireless-configuration"
+            "com.apple.external-accessory.wireless-configuration" \
         )
 
+        # If we change team while resigning, we have no other choice than to use the following entitlements from the PP instead of the App
+        # because they are based on unique identifiers (defined in the developer portal) that can't be shared between teams
+        if [[ "$OLD_TEAM_ID" != "$NEW_TEAM_ID" ]]; then
+            warning "WARNING: Changing team while resigning"
+            warning "WARNING: Using these entitlements from the provisioning profile instead of the existing app:"
+            warning "WARNING: App Groups, Merchant IDs (Apple Pay In-App Payments), iCloud Containers, Pass Type IDs (Wallet)"
+            warning "WARNING: If these capabilities are enabled, make sure AppID and provisioning profile are properly configured"
```

Comment:
> WARNING: If these capabilities are enabled…

Can we check whether the capabilities are enabled and show a more tailored message? Broad, not-necessarily-applicable error messages sometimes confuse and scare devs :)

This could be as simple as printing the `.entitlements` file and telling them what to look for. I also opened Xcodeproj issue https://github.com/CocoaPods/Xcodeproj/issues/562 to make this easier to parse.

Reply:
You're right, but that seemed a good compromise regarding the fact that this issue is preventing people from resigning production apps.

Regarding your last point, Xcodeproj files are not involved here as `--use_app_entitlements` retrieves the entitlements directly from the app binary.
Anyway, IMHO, if someone with no knowledge of the way entitlements work wants to use `resign` with `--use_app_entitlements` option while changing team, they need a tutorial instead of just a few lines in a log to tell them what to look for.